client/src/modules/display_metadata/display_metadata.html

```diff
@@ -0,0 +1,121 @@
+      <li class="title" translate> TREE.DISPLAY_METADATA </li>
+    </ol>
+
+    <div class="toolbar">
+        <div class="toolbar-item">
+            <div uib-dropdown dropdown-append-to-body data-action="open-tools">
+              <a class="btn btn-default" uib-dropdown-toggle>
+                <span class="fa fa-bars"></span> <span class="hidden-xs" translate>FORM.LABELS.MENU</span> <span class="caret"></span>
+              </a>
+    
+              <ul uib-dropdown-menu role="menu" class="dropdown-menu-right">
+                <li role="menuitem">
+                  <a target='_blank'
+                    ng-href="/data_kit/report/?{{ DisplayMetadataCtrl.download(
+                      'pdf', 
+                      DisplayMetadataCtrl.changes, 
+                      DisplayMetadataCtrl.collectorId,
+                      DisplayMetadataCtrl.filterElements,
+                      DisplayMetadataCtrl.patient.uuid,
+                      DisplayMetadataCtrl.patient) }}">
+                    <span class="fa fa-file-pdf-o"></span> <span translate>DOWNLOADS.PDF</span>
+                  </a>
+                </li>
+                <li role="separator" class="divider"></li>    
+                <!-- FIX ME TO DOWMLOAD AS CSV and EXCEL 
+                <li role="menuitem">
+                  <a>
+                    <span class="fa fa-file-excel-o"></span> <span translate>DOWNLOADS.CSV</span>
+                  </a>
+                </li>
+    
+                <li role="menuitem">
+                  <a>
+                    <span class="fa fa-file-excel-o"></span> <span translate>DOWNLOADS.EXCEL</span>
+                  </a>
+                </li> -->
+    
+              </ul>
+            </div>
+          </div>
+
+      <div ng-if="DisplayMetadataCtrl.patientData" class="toolbar-item">
+        <button class="btn btn-default text-capitalize" id="create" ui-sref="display_metadata.patientfill({id : DisplayMetadataCtrl.collector.id, patient: DisplayMetadataCtrl.patient.uuid})" data-method="create">
+          <span class="fa fa fa-file-text"></span> <span class="hidden-xs" translate>FORM.BUTTONS.FILL_FORM</span>
+        </button>
+      </div>
+
+      <div ng-if="DisplayMetadataCtrl.patientData" class="toolbar-item">
+        <button class="btn btn-default text-capitalize" id="create" ui-sref="patientRecord({ patientUuid : DisplayMetadataCtrl.patient.uuid })" data-method="create">
+          <span class="fa fa-medkit"></span> <span class="hidden-xs" translate>FORM.LABELS.MEDICAL_SHEETS</span>
+        </button>
+      </div>
+
+      <div class="toolbar-item">
+        <button
+          ng-click="DisplayMetadataCtrl.search()"
+          data-method="search"
+          class="btn btn-default">
+          <span class="fa fa-search"></span> <span class="hidden-xs" translate>FORM.BUTTONS.SEARCH</span>
+        </button>
+      </div>
+
+      <div class="toolbar-item">
+        <button
```

Comment:
We have a component for this - [bhFilterToggle](https://github.com/IMA-WorldHealth/bhima/blob/master/client/src/js/components/bhFilterToggle.js).  You could use it here to reduce code:

```html
 <bh-filter-toggle on-toggle="DisplayMetadataCtrl.toggleFilter()">
 </bh-filter-toggle>
```

Reply:
I could not use the bh-filter-toggle component just because in normal use the elements that use this component come from the elements defined in advance in the FilterService service, whereas the elements of the structure of the grid comes from the database